samcli/local/apigw/local_apigw_service.py

```diff
@@ -739,6 +739,9 @@ def _request_handler(self, **kwargs):
 
             # invoke the route's Lambda function
             lambda_response = self._invoke_lambda_function(route.function_name, route_lambda_event, tenant_id)
+        except TenantIdValidationError as e:
+            response_data = jsonify({"message": str(e)})
+            endpoint_service_error = make_response(response_data, 400)  # HTTP 400 Bad Request
```

Comment:
It would be better to be consistent with the other Errors and add a new method inside the [`ServiceErrorResponses`](https://github.com/aws/aws-sam-cli/blob/11f52d8e8b7ec65e5b1a9ee80d8853a58964d45a/samcli/local/apigw/service_error_responses.py#L10) class instead of calling `make_response` right here.

But it's just a code refactor that's not critical.